gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/common/AbstractTransportServiceStubClassComposer.java

```diff
@@ -204,20 +220,54 @@ public GapicClass generate(GapicContext context, Service service) {
             .setName(className)
             .setExtendsType(
                 typeStore.get(getTransportContext().classNames().getServiceStubClassName(service)))
-            .setMethods(
-                createClassMethods(
-                    context,
-                    service,
-                    typeStore,
-                    classMemberVarExprs,
-                    callableClassMemberVarExprs,
-                    protoMethodNameToDescriptorVarExprs,
-                    classStatements))
+            .setMethods(methodDefinitions)
             .setStatements(classStatements)
             .build();
     return GapicClass.create(kind, classDef);
   }
 
+  private String getUnsupportedOperationExceptionReason(String callableName, Method protoMethod) {
+    if (protoMethod.stream() == Method.Stream.BIDI
+        || protoMethod.stream() == Method.Stream.CLIENT) {
+      return String.format(
+          "Not implemented: %s(). %s is not implemented for %s",
+          callableName, protoMethod.stream(), Transport.REST);
+    } else if (protoMethod.httpBindings() == null) {
+      return String.format(
+          "Not implemented: %s(). RPC is not enabled for %s", callableName, Transport.REST);
+    } else {
+      return String.format("Not implemented: %s()", callableName);
```

Comment:
What could be in this else branch? 
Also I think a generic error message like `REST transport is not supported for this method yet` should be good enough, it's better than a message that was too generic before. If we really want to tell the differences, we probably want to change the first message to `Not supported:...` instead of `Not implemented:...` since we would never implement REST for BIDI and CLIENT streaming.

Reply:
> What could be in this else branch?

Nothing as of now -- I just intended to give it a default message. I think `REST transport is not supported for this method yet` sounds good. On second thought, perhaps this function can exist in `HttpJsonTransportServiceStubClassComposer`.

> If we really want to tell the differences, we probably want to change the first message to Not supported:... instead of Not implemented:... since we would never implement REST for BIDI and CLIENT streaming.

Yep, I think `Not implemented: ...` makes more sense.